### PR TITLE
Group events by date

### PR DESCRIPTION
**Pull request by charypar**
Add grouping of events by date. I've implemented the data processing as a new selector.

I've also moved the Event type out into a separate file, which seems a lot cleaner, as we're working with it all over the place and pulling it in from the CMS integration module seems a bit wrong. 

I put that module into a `data` folder, which can be a sort of domain model section for functions that operate on the domain entities and the associated types.

## Screenshot

grouping

## Pre-flight check-list

Before raising a pull request

* [x] Documentation
* [x] Unit tests

## Pre-merge check-list

* [ ] Tester approved

### CODE REVIEW

File: src/components/EventList.js [hunk 1]
```diff
@@ -19,10 +27,20 @@ const EventList = ({
...
   onPress
 }: Props) => (
-  <FlatList
+  <SectionList
     contentContainerStyle={styles.container}
-    data={events}
     keyExtractor={event => event.sys.id}
+    sections={events.map(it => ({
+      data: it,
+      title: format(parse(it[0].fields.startTime[locale]), "dddd D MMMM")
```
Comment (frigus02) [inline src/components/EventList.js]: I don't think `parse` is necessay here. You can just pass a string to `format`.

File: src/selectors/events.js [hunk 2]
```diff
@@ -1,5 +1,38 @@
...
+const locale = "en-GB";
+
+const groupByStartTime = (events: Event[]): EventDays => {
+  const sections = events
+    .sort((a: Event, b: Event) => {
+      const diff =
+        parseDate(a.fields.startTime[locale]) -
+        parseDate(b.fields.startTime[locale]);
+
+      return diff / Math.abs(diff);
```
Comment (frigus02) [inline src/selectors/events.js]: Why do we need to divide here?

Comment (charypar) [reply src/selectors/events.js]: The sorting callback expects `-1`, `0` or `1` back. This was a quick way to normalise if the difference is more than one day

File: src/components/EventList.js (hunk 1 above)
Comment (charypar) [reply src/components/EventList.js]: Oh neat! I missed that. Convenient.

File: src/selectors/events.js (hunk 2 above)
Comment (charypar) [reply src/selectors/events.js]: Ah, my bad, it doesn't care about the magnitude, probably thinking about a different language... Even better.

**Comment by charypar** [comment]
@frigus02 Fixed the things you pointed out, also made some changes to the component to make it a bit more readable and added stronger types.

**Comment by frigus02** [comment]
Looks good. Merge it. 😃 

(We should find a way to tell codeclimate that the unit tests are fine.)

**Pull request merged**